tests/integration/test_logcollector/test_reconnect_time/test_reconnect_time.py

```diff
@@ -0,0 +1,123 @@
+configurations_path = os.path.join(test_data_path, 'wazuh_reconnect_time.yaml')
+
+timeout_callback_reconnect_time = 30
+
+parameters = [
+    {'LOCATION': 'Application', 'LOG_FORMAT': 'eventchannel', 'RECONNECT_TIME': '5s'},
+    {'LOCATION': 'Security', 'LOG_FORMAT': 'eventchannel', 'RECONNECT_TIME': '5s'},
+    {'LOCATION': 'System', 'LOG_FORMAT': 'eventchannel', 'RECONNECT_TIME': '5s'},
+    {'LOCATION': 'Application', 'LOG_FORMAT': 'eventchannel', 'RECONNECT_TIME': '40m'},
+    {'LOCATION': 'Security', 'LOG_FORMAT': 'eventchannel', 'RECONNECT_TIME': '40m'},
+    {'LOCATION': 'System', 'LOG_FORMAT': 'eventchannel', 'RECONNECT_TIME': '40m'},
+    {'LOCATION': 'Application', 'LOG_FORMAT': 'eventchannel', 'RECONNECT_TIME': '20h'},
+    {'LOCATION': 'Security', 'LOG_FORMAT': 'eventchannel', 'RECONNECT_TIME': '20h'},
+    {'LOCATION': 'System', 'LOG_FORMAT': 'eventchannel', 'RECONNECT_TIME': '20h'},
+
+]
+metadata = [
+    {'location': 'Application', 'log_format': 'eventchannel', 'reconnect_time': '5s'},
+    {'location': 'Security', 'log_format': 'eventchannel', 'reconnect_time': '5s'},
+    {'location': 'System', 'log_format': 'eventchannel', 'reconnect_time': '5s'},
+    {'location': 'Application', 'log_format': 'eventchannel', 'reconnect_time': '40m'},
+    {'location': 'Security', 'log_format': 'eventchannel', 'reconnect_time': '40m'},
+    {'location': 'System', 'log_format': 'eventchannel', 'reconnect_time': '40m'},
+    {'location': 'Application', 'log_format': 'eventchannel', 'reconnect_time': '20h'},
+    {'location': 'Security', 'log_format': 'eventchannel', 'reconnect_time': '20h'},
+    {'location': 'System', 'log_format': 'eventchannel', 'reconnect_time': '20h'},
+]
+configurations = load_wazuh_configurations(configurations_path, __name__,
+                                           params=parameters,
+                                           metadata=metadata)
+configuration_ids = [f"{x['LOCATION'], x['LOG_FORMAT'], x['RECONNECT_TIME']}" for x in parameters]
+
+
+@pytest.fixture(scope="module", params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def get_local_internal_options():
+    """Get configurations from the module."""
+    return local_internal_options
+
+
+def test_reconnect_time(get_local_internal_options, configure_local_internal_options, get_configuration,
+                        configure_environment, restart_logcollector):
+    """Check if reconnect_time value works properly
+
+    Ensure correspond debug logs are generated when Windows event log service stop. Also, when event log service is
+    restarted, `wazuh-agent` should reconnect to it using reconnect_time value.
+    """
+
+    config = get_configuration['metadata']
+
+    if time_to_seconds(config['reconnect_time']) >=  timeout_callback_reconnect_time:
+        pytest.xfail("Expected fail: ")
```

Comment:
Add more detail here explaining why the error is expected.

Reply:
Done 4d788e12e477bbb1f3655163791253a96b576e07